src/librustc/traits/error_reporting.rs

```diff
@@ -50,7 +50,6 @@ use syntax_pos::{DUMMY_SP, Span};
 impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
     pub fn report_fulfillment_errors(&self,
                                      errors: &[FulfillmentError<'tcx>],
-                                     body_id: Option<hir::BodyId>,
```

Comment:
I take it we're getting the `body_id` from the `ObligationCause` here? I had planned to remove that field once we transition to NLL, since it won't be needed.

Reply:
But I guess we could thread this info back through then.

Reply:
This was only used for the code in `need_type_info.rs` that was visiting the HIR so thankfully I think we won't need it!

I actually couldn't visit the HIR because the `body_id` on the obligation was sometimes "not large enough" to include the closure that had the offending argument.